fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb

```diff
@@ -43,8 +42,10 @@ def run!
         collector.did_raise_error(@program[:name])
         display_user_error!(e, e.message)
       rescue => e # high chance this is actually FastlaneCore::Interface::FastlaneCrash, but can be anything else
-        collector.did_raise_error(@program[:name])
+        collector.did_crash(@program[:name])
         handle_unknown_error!(e)
+      ensure
+        collector.did_finish
```

Comment:
I'm unsure about the consequences of moving this from the `begin` block to here. What has changed? Or was that wrong in the first place